### ISSUE TITLE
RFE: To register TCP:19531,19532 (journald transmission) in "well-known ports" list

### ISSUE THREAD

**Issue by MikhailKasimov** [systemd/systemd#1746]
Not much a technical, just organizational question -- to make a request to IANA to register TCP:19531 (journald transmission) in "well-known ports" list (https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.txt). Suppose, it's good idea for further projects. As for me, as admin, the result of netstat -t: 'Foreign Address somehost:systemd-journald-transmission' will be much more appreciated from  from point of view of auditing of network connections. Yes, I can use -p key to view the PID and program-name on netstat command to have a bit of workaround, but in this case program-name can be another (e.g.firefox). So I understand, that this issue has low-prioritet, but from point of view of auditing of network connections it would be very useful. Thanks!

**Comment by poettering** [comment]
I have no idea how such a registration would work...

**Comment by MikhailKasimov** [comment]
https://www.iana.org/form/ports-services

As I mentioned above , this is an organizational question -- that will mean, that connection on TCP:19531 is _trusted_ and is registered in /etc/protocols as well-known port of, recursively, well-known trusted service (systemd-journald). For example, for dedicated log-collector systems (Logstash, LOGAlyze, Zabbix, etc.) to which we transmit event-logs from journald. Or do you mean something else?

**Comment by poettering** [comment]
Quite frankly, that form has too much legalese for me... i'll leave the registration to somebody else who likes legalese...

**Comment by MikhailKasimov** [comment]
Please, take a look on 1.1.6.a point in https://www.pcisecuritystandards.org/documents/PCI_DSS_v3-1.pdf - "Verify that firewall and router configuration standards 
include a documented list of all services, protocols and ports, 
including business justification for each..." That means that I, as admin, must have a list of well-known services and ports to be in compliance to information security standard on my server. And systemd-journald service cannot be trusted until its port TCP 19531 isn't declared in IANA and, hence, in /etc/protocols. I filed this RFE not because I'm going to tease somebody. It's really needed (not technical) feature. Please, implement it.

**Comment by grawity** [comment]
I agree that IANA reg would be really nice, but I don't see where PCI-DSS implies that as a requirement. It asks for documentation, but doesn't require any specific source, does it?

**Comment by MikhailKasimov** [comment]
In general way -- yes. But auditors are very sensitive, especially when data transmission is going on non-secure channels and especially when transport is not registered. Potentially 19531 can be treated as a leak-channel instead of syslog transmission, which is registered in /etc/services (UDP 514 and UDP 6514 (TLS)). So, IANA reg would be really nice for TCP 19531 (both over HTTP/HTTPS) to avoid a potential risk of treatment 19531 as a leak-channel.

Offtop: To continue speaking about PCI DSS: Due to point 2.2.3, does systemd support SSLv2|3 in its internal http-server?

**Comment by poettering** [comment]
I am fine if the port number is registered at IANA, however, I am not the one who will do the bureaucratic work for that.

We use libminihttpd for the journal-gatewayd stuff. Hence we support whatever that supports.

**Comment by MikhailKasimov** [comment]
IANA #1109127 request is under consideration.

**Comment by MikhailKasimov** [comment]
IANA #1109484 for TCP=19532 is under consideration.

**Comment by alexshpilkin** [comment]
@MikhailKasimov Did anything ever come of those requests? There are no entries for those ports in the protocol registry still, as far as I can see.
